apps/meteor/tests/end-to-end/api/01-users.js

```diff
@@ -1522,31 +1522,6 @@ describe('[Users]', function () {
 				});
 			});
 		});
-
-		function failUpdateUser(name) {
-			it(`should not update an user if the new username is the reserved word ${name}`, (done) => {
-				request
-					.post(api('users.update'))
-					.set(credentials)
-					.send({
-						userId: targetUser._id,
-						data: {
-							username: name,
-						},
-					})
-					.expect('Content-Type', 'application/json')
-					.expect(400)
-					.expect((res) => {
-						expect(res.body).to.have.property('success', false);
-						expect(res.body).to.have.property('error', 'Could not save user identity [error-could-not-save-identity]');
```

Comment:
I wonder if this validation (and others) should still be done "sync" .. 🤔  because the heavy operations we want to avoid using `setImmediate` are all the many MongoDB updates it does that take some time..

so like if we "accept" the update request, it should be done.. it would be weird if we "accept" the update, return a response to the client, but right after that we figure out the name is invalid..

Reply:
I moved the `setImmediate` inside the `saveUserIdentity` and created a flag to decide if the update should run in the background or not to avoid unexpected behavior from other files that use this method.